tests/conftest.py

```diff
@@ -43,106 +139,235 @@ def mock_default_paginated_empty(requests_mock, test_url):
     )
 
 
-@pytest.fixture(autouse=True)
-def mock_yield_channels_0(requests_mock, test_url):
-    url = f"{test_url}/api/paginated/channels?skip=0"
-    mock_resp = {
-        "pagination": {"skip": 0, "limit": 2, "all_records_count": 3},
-        "result": [
-            {
-                "name": "a",
-                "description": "descr a",
-                "private": True,
-                "size_limit": None,
-                "ttl": 36000,
-                "mirror_channel_url": None,
-                "mirror_mode": None,
-                "members_count": 42,
-                "packages_count": 11,
-            },
-            {
-                "name": "b",
-                "description": "descr b",
-                "private": True,
-                "size_limit": None,
-                "ttl": 36000,
-                "mirror_channel_url": None,
-                "mirror_mode": None,
-                "members_count": 42,
-                "packages_count": 11,
-            },
-        ],
-    }
-    requests_mock.get(url, json=mock_resp)
+def live_channels(authed_session, live_server):
+    response = authed_session.get(f"{live_server}/api/channels")
+    assert response.status_code == 200
 
+    channels = [from_dict(Channel, c) for c in response.json()]
+    return channels
 
-@pytest.fixture(autouse=True)
-def mock_yield_channels_2(requests_mock, test_url):
-    url = f"{test_url}/api/paginated/channels?skip=2"
-    mock_resp = {
-        "pagination": {"skip": 2, "limit": 2, "all_records_count": 3},
+
+def get_channel_json(channels, skip, limit):
+    return {
+        "pagination": {
+            "skip": skip,
+            "limit": limit,
+            "all_records_count": len(channels),
+        },
         "result": [
             {
-                "name": "c",
-                "description": "descr c",
-                "private": False,
-                "size_limit": None,
-                "ttl": 36000,
-                "mirror_channel_url": None,
-                "mirror_mode": None,
-                "members_count": 42,
-                "packages_count": 11,
+                "name": channels[i].name,
+                "description": channels[i].description,
+                "private": channels[i].private,
+                "size_limit": channels[i].size_limit,
+                "ttl": channels[i].ttl,
+                "mirror_channel_url": channels[i].mirror_channel_url,
+                "mirror_mode": channels[i].mirror_mode,
+                "members_count": channels[i].members_count,
+                "packages_count": channels[i].packages_count,
```

Comment:
shouldn't there be a way to turn a dataclass into a dict?

Reply:
Ah good point!